pemFioi/quickAlgo/interface-mobileFirst.js

```diff
@@ -249,29 +271,122 @@ var quickAlgoInterface = {
-        if(btn == 'example') {
+        if (btn == 'example') {
             task.displayedSubTask.loadExample()
-        } else if(btn == 'copy') {
+        } else if (btn == 'copy') {
             task.displayedSubTask.blocklyHelper.copyProgram();
-        } else if(btn == 'paste') {
+        } else if (btn == 'paste') {
             task.displayedSubTask.blocklyHelper.pasteProgram();
-        } else if(btn == 'save') {
+        } else if (btn == 'save') {
             task.displayedSubTask.blocklyHelper.saveProgram();
-        } else if(btn == 'restart') {
+        } else if (btn == 'restart') {
             displayHelper.restartAll();
-        } else if(btn == 'best-answer') {
+        } else if (btn == 'edit') {
+            this.openEditExercise();
+        } else if (btn == 'best-answer') {
             displayHelper.retrieveAnswer();
-        } else if(btn == 'blockly-python') {
+        } else if (btn == 'blockly-python') {
             this.displayBlocklyPython();
         }
     },
 
+    openEditExercise: function() {
+        // in python, there are two "exerciseText", we need to selected only the first one
+        // there are two "exerciseText" in python, because we also have a "long" version of the
+        // subject
+        var title = this.userTaskData.title;
+        var subject = this.userTaskData.subject;
+        var editExerciseHtml = "<div class=\"content connectPi qpi\">" +
+            "    <div class=\"panel-heading\">" +
+            "        <h2 class=\"sectionTitle\">" +
+            "            <span class=\"iconTag\"><i class=\"icon fas fa-pencil-alt\"></i></span>" +
+                         this.strings.editWindowTitle +
+            "        </h2>" +
+            "    <div class=\"exit\" id=\"editclose\"><i class=\"icon fas fa-times\"></i></div>" +
+            "    </div>" +
+            "    <div class=\"panel-body\">" +
+            "        <div id=\"editExerciseTitle\">" +
+            "            <label>" + this.strings.titleEdition + "</label><input id=\"editExerciseTitleInput\" type=\"text\" value=\"" + title + "\"/>" +
+            "        </div>" +
+            "        <div id=\"editExerciseDescription\">" +
+            "            <label>" + this.strings.descriptionEdition + "</label>" +
+            "            <textarea rows=\"10\" id=\"editExerciseDescriptionTextarea\">" + subject + "</textarea>" +
+            "        </div>" +
+            "        <div id='panel-body-bottom'>" +
+            "            <button id='saveExerciseChanges'>" + this.strings.saveAndQuit + "</button>" +
+            "        </div>" +
+            "    </div>" +
+            "</div>";
+
+        window.displayHelper.showPopupDialog(editExerciseHtml);
+
+        var that = this;
+
+        $("#editclose").click(function() {
+            var newTitle = $("#editExerciseTitleInput").val();
+            var newDesc = $("#editExerciseDescriptionTextarea").val();
+            var oldTitle = document.title;
```

Comment:
You can load oldTitle and oldDescription from this.userTaskData now, rather than reread from the task itself.